backend/routes/admin.py

```
@@ -886,3 +897,769 @@ async def get_all_users(session: Dict[str, Any] = Depends(require_admin_role)) -
+
+        # Prevent deactivating categories that have questions
+        if category_data.is_active is False:
+            # Get current settings to check for questions
+            settings_json = admin_db_manager.get_admin_setting("followup_settings")
+            if settings_json:
+                settings = FollowUpSettings.from_json(settings_json)
+                if existing["name"] in settings.custom_questions and settings.custom_questions[existing["name"]]:
+                    raise HTTPException(
+                        status_code=400,
+                        detail=f"Cannot deactivate category '{existing['display_name']}' - it contains questions. Remove questions first.",
+                    )
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)